modules/ding_seo/ding_seo.module

```diff
@@ -226,3 +228,185 @@ function ding_seo_get_person_nodes(array $names) {
+      variable_set('ding_seo_sitemap_last_check', time());
+    }
+  } catch (Exception $exception) {
+    watchdog_exception('ding_seo', $exception);
+  }
+
+}
+
+/**
+ * Get the new links from last update.
+ */
+function ding_seo_get_objects($last_check ) {
+  $objects = [];
+
+  $query = db_select('ting_object_sitemap', 't')
+    ->condition('t.created', $last_check , '>')
+    ->fields('t', ['tid', 'ting_object_id'])
+    ->execute();
+
+  while ($value = $query->fetchAssoc()) {
+    $tid = $value['tid'];
+    $objects[$tid ] = $value['ting_object_id'];
+  }
+
+  return $objects;
+}
+
+/**
+ * Insert the new links into xmlsitemaptable.
+ */
+function ding_seo_update_sitemap_table($objects) {
+  $maxid = (int) db_query("SELECT max(id) FROM {xmlsitemap} WHERE type = 'ting_object'")->fetchField() ?: 0;
+  $chunks = array_chunk($objects, 1000, TRUE);
+  foreach ($chunks as $chunk) {
+    // We could use xmlsitemap_link_save but it takes only one link at a time. For performance reasons
+    // we update the database table our selves.
+    $query = db_insert('xmlsitemap')->fields(array('id', 'type', 'loc', 'language', 'priority'));
+    foreach ($chunk as $id => $object) {
+      //If our id isn't the largest somthings wrong.
+      if ($id > $maxid) {
+        $query->values(
+          [
+            'id' => $id,
+            'type' => 'ting_object',
+            'loc' => 'ting/object/' . $object,
+            'language' => 'und',
+            'priority' => 0.5,
+          ]
+        );
+      }
+    }
+    $query->execute();
+  }
+}
+
+/**
+ * Checks wether to put the ting_object into a sitemaps.
+ *
+ *  We only want ting_objects from the Library's own collection.
+ *  And we only want the ones the users have viewed as object or collection.
+ */
+function ding_seo_check_object_for_sitemap($ting_object, $view_mode) {
+  if ($view_mode != 'full' && $view_mode != 'collection_list') {
+    return;
+  }
+
+  if ($ting_object->getAc_source() == 'Bibliotekskatalog') {
+    try {
+      db_merge('ting_object_sitemap')
+        ->insertFields([
+          'ting_object_id' => $ting_object->getId(),
+          'created' => time(),
+          'changed' => time(),
+        ])
+        ->updateFields([
+          'changed' => time(),
+        ])
+        ->key(['ting_object_id' => $ting_object->getId()])
+        ->execute();
+    } catch (Exception $exception) {
+      watchdog_exception('ding_seo', $exception);
+    }
+  }
+}
+
+/**
+ * Check if our ting_object_sitemap table exceedes the maximum number of link.
+ *
+ * We trim the links which haven't been visited for the longest time.
+ */
+function ding_seo_check_sitemap_size() {
+  $max_size = variable_get('ding_seo_maximum_links_in_sitemap', 50000);
+  $size = (int) db_query("SELECT count(*) FROM {ting_object_sitemap}")->fetchField() ?: 0;
+
+  if ($size > $max_size) {
+    $number_to_delete = variable_get('ding_seo_number_of_links_to_delete', 1000);
```

Comment:
Have you considered ditching this setting and just deleting `$size - $max_size` (if positive) every time?